api/envoy/type/matcher/regex.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package envoy.type.matcher;
+
+option java_outer_classname = "StringProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type.matcher";
+option go_package = "matcher";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: RegexMatcher]
+
+// A regex matcher designed for safety when used with untrusted input.
+message RegexMatcher {
+  // Google's `re2 <https://github.com/google/re2>`_ regex engine. The regex string must adhere to
+  // the documented `syntax <https://github.com/google/re2/wiki/Syntax>`_. The engine is designed
+  // to complete execution in linear time as well as limit the amount of memory used. In the future
+  // different aspects of the engine may be made configurable.
+  message GoogleReEngine {
```

Comment:
Can we call this RE2 or maybe GoogleRE2? There are actually a few different REs used throughout Google though probably most people know what you mean.